src/hooks/useExtraAPY.ts

```diff
@@ -1,20 +1,107 @@
 import Decimal from "decimal.js";
 
-import { getTotalSupplyAndBorrowUSD } from "../redux/assetsSelectors";
+import { getAccountPortfolio } from "../redux/accountSelectors";
+import { getAssets, getTotalSupplyAndBorrowUSD } from "../redux/assetsSelectors";
+import { computeDailyAmount } from "../redux/selectors/getAccountRewards";
+import { getStaking } from "../redux/selectors/getStaking";
+import { getConfig } from "../redux/appSelectors";
 import { useAppSelector } from "../redux/hooks";
+import { shrinkToken } from "../store/helper";
 
-export function useExtraAPY({ tokenId, isBorrow }) {
-  const [totalSupplyUSD, totalBorrowUSD] = useAppSelector(getTotalSupplyAndBorrowUSD(tokenId));
+export function useExtraAPY({ tokenId: assetId, isBorrow }) {
+  const [totalSupplyUSD, totalBorrowUSD] = useAppSelector(getTotalSupplyAndBorrowUSD(assetId));
+  const { xBRRR, extraXBRRRAmount } = useAppSelector(getStaking);
+  const portfolio = useAppSelector(getAccountPortfolio);
+  const appConfig = useAppSelector(getConfig);
+  const assets = useAppSelector(getAssets);
+  const asset = assets.data[assetId];
+  const assetDecimals = asset.metadata.decimals + asset.config.extra_decimals;
 
-  const computeRewardAPY = (rewardsPerDay, decimals, price) => {
-    return new Decimal(rewardsPerDay)
+  const computeRewardAPY = (rewardTokenId, rewardsPerDay, decimals, price) => {
+    const rewardAsset = assets.data[rewardTokenId];
+    const type = isBorrow ? "borrowed" : "supplied";
+    const farmData = portfolio.farms?.[type]?.[assetId]?.[rewardTokenId];
+
+    const totalDailyRewards = new Decimal(rewardsPerDay)
       .div(new Decimal(10).pow(decimals))
-      .mul(365)
-      .mul(price)
-      .div(isBorrow ? totalBorrowUSD : totalSupplyUSD)
-      .mul(100)
       .toNumber();
+
+    const totalDeposits = isBorrow ? totalBorrowUSD : totalSupplyUSD;
```

Comment:
Shouldn't it be the same as what is used in the computeStakingRewardAPY? As below:    

const totalAssetUSD = isBorrow
      ? totalBorrowAssetUSD
      : totalSupplyAssetUSD + totalCollateralAssetUSD;